ibtk/include/ibtk/snapshot_utilities.h

```diff
@@ -0,0 +1,63 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (c) 2022 - 2022 by the IBAMR developers
+// All rights reserved.
+//
+// This file is part of IBAMR.
+//
+// IBAMR is free software and is distributed under the 3-clause BSD
+// license. The full text of the license can be found in the file
+// COPYRIGHT at the top level directory of IBAMR.
+//
+// ---------------------------------------------------------------------
+
+/////////////////////////////// INCLUDE GUARD ////////////////////////////////
+
+#ifndef included_IBTK_snapshot_utilities
+#define included_IBTK_snapshot_utilities
+
+/////////////////////////////// INCLUDES /////////////////////////////////////
+
+#include <ibtk/SnapshotCache.h>
+#include <ibtk/ibtk_utilities.h>
+
+#include "CellVariable.h"
+#include "HierarchyDataOpsReal.h"
+#include "IntVector.h"
+#include "SideVariable.h"
+#include "tbox/Database.h"
+#include "tbox/Pointer.h"
+
+#include <string>
+#include <vector>
+
+/////////////////////////////// CLASS DEFINITION /////////////////////////////
+
+namespace IBTK
+{
+/*!
+ * This function interpolates in time between two snapshots. We find the two closest time points stored in the snapshot
+ * cache and linear interpolate in time between them. Each snapshot will be transferred onto the provided hierarchy. A
+ * scratch patch index with the same patch layout as u_idx must be provided, and data for both must be allocated. The
+ * refine_type must be a valid refinement operation for the data layout.
+ *
+ * This assumes that the time is between two snapshots stored in the cache. An error will occur if two time points can
+ * not be found or if there is a single snapshot, if the time value does not match what's stored in the cache.
+ *
+ * The hier_data_ops object, if provided, must match the same variable type used in for the snapshots.
```

Comment:
Should this class handle time periodic cases?

Reply:
This function just interpolates between two snapshots. Although, as this is written, we wouldn't be able to interpolate between the last and first snapshot. We'll need a function that knows about periodicity.